src/190B20.c

```diff
@@ -1157,9 +1157,49 @@ s32 count_power_plus(s32 arg0) {
     return pp;
 }
 
-INCLUDE_ASM(s32, "190B20", deduct_current_move_fp);
+void deduct_current_move_fp(void) {
+    BattleStatus* battleStatus = &gBattleStatus;
+    PlayerData* playerData = &gPlayerData;
+    Actor* actor = battleStatus->playerActor;
+    s32 fpCost = gMoveTable[battleStatus->selectedMoveID].costFP;
+
+    if (fpCost != 0) {
+        fpCost -= player_team_is_ability_active(actor, ABILITY_FLOWER_SAVER);
+        fpCost -= player_team_is_ability_active(actor, ABILITY_FLOWER_FANATIC) * 2;
+        if (fpCost < 1) {
+            fpCost = 1;
+        }
+    }
+
+    playerData->curFP = playerData->curFP - fpCost;
+}
+
+void reset_actor_turn_info(void) {
+    BattleStatus* battleStatus = &gBattleStatus;
+    Actor* actor;
+    s32 i;
+
+    for (i = 0; i < ARRAY_COUNT(battleStatus->enemyActors); i++) {
+        actor = battleStatus->enemyActors[i];
+        if (actor != NULL) {
+            actor->hpChangeCounter = 0;
+            actor->damageCounter = 0;
+            actor->unk_204[0] = 0;
+        }
 
-INCLUDE_ASM(s32, "190B20", reset_actor_turn_info);
+    }
+    actor = battleStatus->playerActor;
```

Comment:
Do you actually have to keep saving the current actor into this temp? You can't just do battleStatus->playerActor-> whatever = 0?

Reply:
nm. tried it and you can't.